src/plugins/controls/server/options_list/options_list_queries.ts

```diff
@@ -139,6 +142,57 @@ const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } =
       ),
   },
 
+  /**
+   * the "IP" query / parser should be used when the options list is built on a field of type IP.
+   */
+  ip: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => {
```

Comment:
Note that I opted to do everything with an `ip_range` aggregation here.

In my original version of this code (as can be seen in b78e574fcf0ee65ff607f89989ab8cb0be1b3459), I had three different types of aggregations for each of the three possible search string cases (i.e. no search string, partial search string, full IP search string) - while this is marginally more efficient than using an `ip_range` aggregation for every case, it made for messier code for both `buildAggregation` and `parse`.